napari/layers/labels/labels.py

```diff
@@ -366,6 +370,41 @@ def _set_view_slice(self):
         self._update_coordinates()
         self.events.set_data()
 
+    def _set_view_slice(self):
+        """Sets the view given the indices to slice with."""
+        self._set_view_slice_keep_history()
+        self._undo_history = []
+        self._redo_history = []
+
+    def _trim_history(self):
+        while (
+            len(self._undo_history) + len(self._redo_history)
+            > self._history_limit
+        ):
+            del self._undo_history[0]
```

Comment:
Maybe use a `deque` instead of a list? You get some nice things (such as `maxlen`) for free:

https://docs.python.org/3.7/library/collections.html#collections.deque

Reply:
great suggestions, deques look super awesome! since we're checking the length of two stacks, `maxlen` probably won't work atm, but will be super useful when we implement #474 (along with `rotate`)!